### PR TITLE
support old browsers without polyfills

### PR DESCRIPTION
**Pull request by graingert**
### Reasons for making this change

Currently react-jsonschema-form forces downstream projects to use babel-polyfill, and includes duplicated babel helpers.

This removes those duplicate helpers and adds ponyfills for modern globals like Set, Promise and Objects.assign.

This PR also replaces methods like `Array.prototype.includes` and `Array.prototype.fill` with ponyfills from `core-js`

If this is related to existing tickets, include links to them as well:

#798

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've checked the rendering of the Markdown text I've added
  - [ ] If I'm adding a new section, I've updated the Table of Content
* [x] **I'm adding or updating code**
  - [ ] I've added and/or updated tests
  - [ ] I've updated docs if needed
  - [x] I've run `npm run cs-format` on my branch to conform my code to [prettier](https://github.com/prettier/prettier) coding style
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

File: src/components/widgets/SelectWidget.js [hunk 1]
```diff
@@ -3,18 +3,16 @@ import PropTypes from "prop-types";
 
 import { asNumber } from "../../utils";
 
+const nums = new Set(["number", "integer"]);
```
Comment (magopian) [inline src/components/widgets/SelectWidget.js]: Shouldn't this `Set` be taken from `core-js`? Or maybe the same change as above, using the polyfilled `includes` from `core-js`.

**Comment by graingert** [comment]
transform-runtime automatically ponyfills globals like Set Promise setImmediate and Object.assign

**Comment by magopian** [comment]
oh, wait, no, I borked it. Let me try again...

**Comment by graingert** [comment]
@magopian let me take over

File: package.json [hunk 2]
```diff
@@ -42,10 +42,10 @@
   },
   "dependencies": {
     "ajv": "^5.2.3",
-    "babel-preset-env": "^1.7.0",
```
Comment (graingert) [inline package.json]: woops, I put this in the wrong place in my last pr, should have been a devDep

**Comment by graingert** [comment]
@magopian done

**Comment by glasserc** [comment]
Semver says it's a minor bump only if new functionality is added *to the API*. The API hasn't changed, so it's a patch version.

**Comment by glasserc** [comment]
@graingert , could you please look at #991 and #1082 and tell me what you think? I am worried that we may have broken the way some of our users are including rjsf, but I'm not an expert in JS packaging so I'm hoping you can help.

**Comment by graingert** [comment]
Ok I'll have a look tomorrow

**Comment by graingert** [comment]
@glasserc I took a look - looks like it's a simple fix to the webpack babel-loader config to exclude babel loader on babel and core-js in webpack.

This is why it works when people use the package from npm directly

**Comment by glasserc** [comment]
To elaborate on the above comment, I spoke with @graingert via IRC and he explained that he thought the problem was that we're running babel on all our dependencies, including the ones that provide ponyfills such as corejs and babel-runtime, and that this is due to a webpack misconfiguration. I'll try to fix it and open a PR today.

**Comment by graingert** [comment]
@glasserc currently I think you only *need* to exclude `node_modules\/core-js\/.*` from `babel-loader`

It may be prudent to switch to a whitelist, however
